src/components/menu/nz-submenu.component.ts

```diff
@@ -58,26 +59,35 @@ import { debounceTime } from 'rxjs/operators/debounceTime';
 })
 
 export class NzSubMenuComponent implements OnInit, OnDestroy, AfterViewInit {
+  private _open = false;
   isInDropDown = false;
   level = 1;
   _$mouseSubject = new Subject();
   @ContentChildren(NzSubMenuComponent) subMenus;
-  @Input() nzOpen = false;
   @Output() nzOpenChange: EventEmitter<boolean> = new EventEmitter();
 
+  @Input()
+  set nzOpen(value: boolean) {
+    this._open = toBoolean(value);
+  }
+
+  get nzOpen(): boolean {
+    return this._open;
+  }
+
   get subItemSelected(): boolean {
-    return !!this.nzMenuComponent.menuItems.find(e => e.selected && e.nzSubMenuComponent === this);
```

Comment:
It was accessing internal field originally, added type info for `menuItems` to allow type checking.